pandas/plotting/_matplotlib/hist.py

```diff
@@ -243,15 +244,26 @@ def _grouped_hist(
     sharey : bool, default False
     rot : int, default 90
     grid : bool, default True
+    legend: : bool, default False
     kwargs : dict, keyword arguments passed to matplotlib.Axes.hist
 
     Returns
     -------
     collection of Matplotlib Axes
     """
+    if legend:
+        assert "label" not in kwargs
```

Comment:
Is pandas the only one calling this, or could users call it with labels in kwargs?

Users shouldn't see bare asserts like this. It should be a `ValueError`.

Reply:
And just to verify the expected behavior, if the user provides a `label` why wouldn't we want to use that? Can we just do `kwargs.setdefault("label", data.name)` (or the other cases)?